### ISSUE TITLE
[FEA] Pruning from dense input

### ISSUE THREAD

**Issue by afender** [rapidsai/cugraph#1247]
Users have reported using dense input such as euclidian distance matrices.
This is problematic because cuGraph is a sparse framework.

To address that we propose adding a pruning feature that accepts a dense matrix and returns only the values below/above a threshold into a sparse structure that can then be used to construct a cugraph `Di/Graph` efficiently.

`coo_ gdf = prune(dense_dense_input, min_threshold, max_threshold) `

**Comment by Sh4zKh4n** [comment]
@afender its funny, I am trying to work out how to use cugraph for on of my own areas. Which is I have a large dense voxel array and I want to work out a number of paths from a source vertex to other vertices preferably of a different segmentation. I just dont care about 0 label in a uint8 image. I havent worked that out it all yet but the code I am trying to work through right now is as follows (ultimately I would look to take this via dask into cuDF to build a graph with cugraph). Its a work in progress. The sparse library maintainers think the array would densify if taken to pd's but i am not sure. especially if you call .asarray. Just thought I would i would share how i am trying to get around this problem may be you can build or tell me it's terrible.

## Option 1: Voxels with 0 (threshold value) are ignored and changed into a point cloud dataframe.
```python
import gradslam as gs
import numpy as np

# create basic voxel grid
voxel_grid = np.zeros((10, 10, 10, 1), dtype=np.uint8)
voxel_grid[4:6, 4:6, 4:6] = 100

# convert voxel grid to xyz coordinates and features
nonzero_inds = np.nonzero(voxel_grid)[:-1]
points = torch.tensor(nonzero_inds, dtype=torch.float32).T
features = torch.tensor(voxel_grid[nonzero_inds])

# construct gradslam.Pointclouds object
pointclouds = gs.Pointclouds(points=[points], features=[features])
```

## option 2: using the sparse library. they use a coo format 

so it would look like this in the end:
```python
import numpy as np
import sparse #edit forgot to import sparse library in original post.
import dask.array as da
import networkx as nx
import dask.dataframe as dd
import pandas as pd

big_array =da.random.random((15000, 2500, 10000), chunks=(1000, 500, 1000), dtype=np.uint8) # big array 
big_array[big_array<0.9] = 0 # fill most of the array with zeros

ba_s = sparse.COO(big_array)


bas_df = dd.io.from_dask_array(ba_s, columns=['z', 'y', 'x', 'label'])

bas_df.to_hdf('df_sparse_BA.hdf', '/data')    

bas_df = dd.read_hdf('df_sparse_BA.hdf') 
object_id =0

def point_graph(df, object_id=object_id):
    # create kdtree
    tree = spatial.cKDTree(df[['z', 'y', 'x']])

    # get neighbours within distance for every point, store in dataframe as edges
    edges = pd.DataFrame({'src':[], 'tgt':[]}, dtype=int)
    for source, target in enumerate(tree.query_ball_tree(tree, r=2)):
        target.remove(source)
        if target:
            edges = edges.append(pd.DataFrame({'src':[source] * len(target), 'tgt':target}), ignore_index=True)

    # create graph for points using edges from Balltree query
    G = nx.from_pandas_dataframe(edges, 'src', 'tgt')

    for i in sorted(G.nodes()):
        G.node[i]['label'] = nodes.label[i]
        G.node[i]['x'] = nodes.x[i]
        G.node[i]['y'] = nodes.y[i]
        G.node[i]['z'] = nodes.z[i]
   return G

NN_graph = point_graph(df)
```
